app/controllers/user_controller.php

```diff
@@ -73,4 +84,98 @@ public function logout()
         session_destroy();
         redirect('user/login');
     }
+
+    public function profile()
+    {
+        $user = new User();
+        $user->id = Param::get('user_id');
+        $user->getProfile();
+
+        if(!isset($user->username)) {
+            redirect('notfound/pagenotfound');
+        }
+
+        $threads_followed = array();
+        $thread_followed_id = Follow::getFollowedThreadIds($user->id);
+
+        foreach($thread_followed_id as $thread) {
+            $threads_followed[] = Thread::getById($thread['thread_id']);
+        }
+
+        foreach ($threads_followed as $thread) {
+            $thread->username = User::getUsernameById($thread->user_id);
+        }
+
+        $threads_created = Thread::getByUserId($user->id);
+
+        foreach ($threads_created as $thread) {
+            $thread->username = User::getUsernameById($thread->user_id);
+        }
+
+        $this->set(get_defined_vars());
+    }
+
+    public function edit()
+    {
+        $process = Param::get('process', 'edit');
+        $user = new User();
+
+        switch($process) {
+            case self::EDIT_ACCOUNT:
+                $user->id = $_SESSION['userid'];
```

Comment:
I think you should get the user first to check if s/he actually exists before allowing him/her to edit info.
Simply setting the $user->id attribute with the value stored in session is not good. 
What if the user is deleted in the DB but his ID is still stored in session po? :sweat:
Please refer to the ff example:
```php
try {
    // will throw RecordNotFoundException if user is not found
    $user = User::get($_SESSION['userid']); 

    $user->setFirstName(Param::get('firstname'));
    $user->update();
} catch (RecordNotFoundException $e) {
    $page = 'error_page'; // or error message
}
...
```